### PR TITLE
Support updating programs via PATCH requests

### PR DESCRIPTION
**Pull request by rlucioni**
ECOM-2752.

@jimabramson, please review. @AlasdairSwan, FYI. This will require the authoring app to submit data only for fields being modified. For example, if the subtitle is changed from "foo" to "bar," the API will expect a PATCH request with the following payload:
```javascript
{'subtitle': 'bar'}
```

### CODE REVIEW

**Comment by AlasdairSwan** [comment]
@rlucioni because this PR will break functionality until the front end has been updated would it make sense to hold off on merging it until the PR for the front end update is ready to go too?

**Comment by rlucioni** [comment]
@AlasdairSwan sure, that makes sense. I should note that the PATCH handler added here is capable of accepting request payloads with multiple keys, if multiple fields are to be changed. For example:
```javascript
{
    'name': 'baz',
    'subtitle': 'bar'
}
```
However, since the front end currently includes fields like `organizations` and `course_codes` in its request payloads, things would still break. I'll hold off on merging this until the front end no longer submits those values in PATCH requests.

File: programs/apps/api/serializers.py [hunk 1]
```diff
@@ -93,6 +93,17 @@ def create(self, validated_data):
...
+    def update(self, instance, validated_data):
+        """
+        Update a program.
+        """
+        for field, value in validated_data.viewitems():
+            setattr(instance, field, value)
+
+        instance.save()
+
+        return instance
```
Comment (jimabramson) [inline programs/apps/api/serializers.py]: @rlucioni question; if a client submits nested objects as part of the PATCH request will this cover those updates as well?

Comment (jimabramson) [inline programs/apps/api/serializers.py]: never mind, you answered my question above.

**Comment by jimabramson** [comment]
:+1:

**Comment by rlucioni** [comment]
Turns out I unwittingly re-implemented DRF's [default update method](https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/serializers.py#L871-L882). I will scrap these changes but leave the updated, more correct tests in place.

**Pull request merged**